File: lib/data/local/model/event.dart
import 'package:drift/drift.dart';

class Event extends Table {
  @override
  String? get tableName => 'FEvent';

  IntColumn get id => integer()();
  IntColumn get categoryId => integer()();
  TextColumn get title => text()();
  DateTimeColumn get date => dateTime()();
  TextColumn get note => text().nullable()();
  TextColumn get audio => text().nullable()();
  TextColumn get shortDesc => text().nullable()();
  BoolColumn get isDraft => boolean()();
  BoolColumn get isArchive => boolean()();
  IntColumn get musicGroupId => integer().nullable()();
  TextColumn get video => text().nullable()();

  @override
  Set<Column> get primaryKey => {id};
}

File: composeApp/src/commonMain/kotlin/by/geth/gethsemane/domain/model/Event.kt
package by.geth.gethsemane.domain.model

import kotlinx.datetime.LocalDateTime

data class Event(
    val id: Int,
    val categoryId: Int,
    val musicGroupId: Int?,
    val title: String,
    val dateTime: LocalDateTime,
    val isDraft: Boolean,
) {
    companion object {
        const val WORSHIP_CATEGORY_ID = 10
    }
}
